### ISSUE TITLE
Is this model only works well for swin backbone??

### ISSUE THREAD

**Issue by gmk11** [Gengzigang/PCT#4]
I tried to change the backbone to see the performances of the model , i retrained my backbone(Resnet) with heatmap supervision on coco as said in the doc of the repository and then trained the tokenizer ,  and  the Classifier but i get bad Precision for the final model (AP = 0.150 and AP@0.5 = 0.4) is there something i missed ? or just the model works only for swin?

**Comment by Gengzigang** [comment]
We have not tried using ResNet as a backbone. Could you provide more details on the performance of each of your three steps (heatmap, tokenizer, classifier)?

**Comment by gmk11** [comment]
@Gengzigang ???

**Comment by Gengzigang** [comment]
If you want to use ResNet as the backbone and train the classifier using a pre-trained tokenizer, there should be no problems. You can choose to directly load the backbone-agnostic tokenizer we provided (https://github.com/Gengzigang/PCT#remove-image-guidance), which has been trained without image guidance, to train your classifier.

**Comment by Gengzigang** [comment]
Regarding what you mentioned about a better decoder, our current experiments have shown that the decoder we are currently using is already sufficient. If the classification is completely accurate, the average precision (AP) can reach 99.

**Comment by gmk11** [comment]
thanks , what do you think about the results of my heatmap ? is it sufficient?

**Comment by Gengzigang** [comment]
You can directly use the ResNet backbone of Simple Baseline (https://arxiv.org/abs/1804.06208). MMPose has supported it.

**Comment by Gengzigang** [comment]
Yes, our backbone is well-trained.

**Comment by gmk11** [comment]
thanks for your time and responses

**Comment by 15023520700** [comment]
> 我检查了我的代码，发现热图权重没有为我的骨干启动，我现在正在对分词器和分类器进行新的训练。我在热图训练中获得了以下性能： 平均精度 （AP） @[ IoU=0.50：0.95 | area= all | maxDets= 20 ] = 0.412 平均精度 （AP） @[ IoU=0.50 | area= all | maxDets= 20 ] = 0.760
> 
> 对于分词器： 平均精度 （AP） @[ IoU=0.50：0.95 | area= all | maxDets= 20 ] = 0.843 平均精度 （AP） @[ IoU=0.50 | area= all | maxDets= 20 ] = 0.975
Excuse me, how do you enable custom backbone network weights

**Comment by ll8657** [comment]
> I checked my code and saw that the heatmap weights weren't initilalized for my backbone, i am doing a new training of the tokenizer and the classifier right now. i got this performance for my heatmap training: Average Precision (AP) @[ IoU=0.50:0.95 | area= all | maxDets= 20 ] = 0.412 Average Precision (AP) @[ IoU=0.50 | area= all | maxDets= 20 ] = 0.760
> 
> this for the tokenizer : Average Precision (AP) @[ IoU=0.50:0.95 | area= all | maxDets= 20 ] = 0.843 Average Precision (AP) @[ IoU=0.50 | area= all | maxDets= 20 ] = 0.975

Excuse me, how do you enable custom backbone network weights

**Comment by gmk11** [comment]
> area

I created a backbone like the implementation of the swin backbone , and I uploaded my weight in it like when you use a resnet  backbone from PyTorch , and I désactivated the initial upload weight function when I trained the heatmap and tokenizer and reactivated when I trained the classifier. 

You can create for example , create a Resnet class in the same file where swin class is implemented then you can  replace the backbone  in the file where backbone= ‘swin’ by backbone = ‘Resnet’

hope it help

**Comment by ll8657** [comment]
> > area
> 
> I created a backbone like the implementation of the swin backbone , and I uploaded my weight in it like when you use a resnet backbone from PyTorch , and I désactivated the initial upload weight function when I trained the heatmap and tokenizer and reactivated when I trained the classifier.
> 
> You can create for example , create a Resnet class in the same file where swin class is implemented then you can replace the backbone in the file where backbone= ‘swin’ by backbone = ‘Resnet’
> 
> hope it help

Thanks for your warm reply, I have now replaced a custom backbone network and trained on the heat map supervision with the final accuracy AP=0.74. Since I don't see the code to import weights, if I want to train tokenizer and classifiers with this weight, I just need to modify Backbone='hmp_backbone' in the config file. Will Pytorch automatically import my heat map monitor weights

**Comment by JasOleander** [comment]
@Gengzigang 
Hello, could you tell me about training time with and without image guidance for tokenizer?  I didn't see that in your paper. And, what is the output of decoder (integer or logits) for classifier you mentioned in figure 1?

**Comment by JasOleander** [comment]
> I checked my code and saw that the heatmap weights weren't initilalized for my backbone, i am doing a new training of the tokenizer and the classifier right now. i got this performance for my heatmap training: Average Precision (AP) @[ IoU=0.50:0.95 | area= all | maxDets= 20 ] = 0.412 Average Precision (AP) @[ IoU=0.50 | area= all | maxDets= 20 ] = 0.760
> 
> this for the tokenizer : Average Precision (AP) @[ IoU=0.50:0.95 | area= all | maxDets= 20 ] = 0.843 Average Precision (AP) @[ IoU=0.50 | area= all | maxDets= 20 ] = 0.975

Hello. Did you train hmp_base.py file again with pretrained 'ResNet model weights'? 
Or tokenizer.py file with pretrained 'ResNet Heatmap' weights?
